llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -172,15 +177,17 @@ def get_validation_collate_fn(self):
 
         return None
 
-    def postprocess_batch_predictions(
-        self, cfg: Any, input_batch: Dict, output: Dict
-    ) -> Dict:
+    def postprocess_batch_predictions(self, cfg: Any, output: Dict) -> Dict:
         predicted_text = [
             self.tokenizer.decode(ids, skip_special_tokens=True).strip()
             for ids in output["predicted_answer_ids"]
         ]
         output["predicted_text"] = np.array(predicted_text)
-        del output["predicted_answer_ids"]
+
+        if not cfg.training.use_rlhf:
```

Comment:
Do you think it makes sense to repopulate `predicted_answer_ids` in `train.py` instead?

Reply:
You mean making a copy of the `predicted_answer_ids` first and then add back to the dict after the post processing? sounds a bit complicated to me, or I am probably missing something.

Reply:
Should be fine the way it is currently